src/arrayMethodJoin.js

```diff
@@ -4,9 +4,31 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = (',')) {
+    let result = '';
+
+    if (!this.length) {
+      return '';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      if (i === this.length - 1) {
+        result += nullCheck(this[i]);
+      } else {
+        result += nullCheck(this[i]) + separator;
+      }
+    }
+
+    return result;
   };
+
+  function nullCheck(value) {
+    if (value === null || value === undefined) {
+      return '';
+    }
+
+    return value.toString();
```

Comment:
If I'm not mistaken when you're trying to write something into a string it implicitly converts to string so there's no need to convert value to string. This process is going here:
```js
result += nullCheck(this[i]);
```

Reply:
Not so sure about this at least didn't work for me